packages/ember-toucan-core/src/components/form/fields/checkbox.hbs

```diff
@@ -25,14 +25,39 @@
         />
 
         <div class="flex flex-col">
-          <span
-            class="type-md-tight block leading-4
-              {{if @isDisabled 'text-disabled' 'text-titles-and-attributes'}}"
-            data-label
-          >{{@label}}</span>
+          {{this.assert
+            (hash
+              blockExists=(has-block "label")
+              argName="label"
+              arg=@label
+              required=true
+            )
+          }}
+          {{#if (this.has (has-block "label") @label)}}
```

Comment:
I'm kind of wondering if we can combine these two helpers into one so that there's less logic here.  Essentially something like the following, without the need for `this.assert` plus the `{{#if (this.has` pieces:

```hbs
{{#if (this.assertBlockOrLabel (hash blockExists=(has-block "label") name="label" argument=@label) isRequired=true)}}
```

and the code for that being something like:

```js
// in the backing ts class

// I don't think this needs to go in `/helpers` as it's more of a shared function than a true Ember helper
import { assertBlockOrArgumentExists } from '../../../-private/assertBlockOrArgumentExists';
import type { AssertBlockOrArg }  from '../../../-private/assertBlockOrArgumentExists';

// ... Other stuff ...

assertBlockOrLabel = ({ blockExists, argument, name, isRequired }: AssertBlockOrArg) =>
    assertBlockOrArgumentExists({ blockExists, argument, name, isRequired });
```


```ts
import { assert } from '@ember/debug';

export type AssertBlockOrArg = {
  /**
   * The name of the argument that should be required.
   */
  argument?: string;
  /**
   * The name of the block or argument. Currently, the block and argument name must match.
   */
  name: string;
  /**
   * Does the provided block exist?
   */
  blockExists: boolean;
  /**
   * Should this assertion be required, or is it optional?
   */
  isRequired?: boolean;
};

export const assertBlockOrArgumentExists = ({
  argument,
  name,
  blockExists,
  isRequired
}: AssertBlockOrArg) => {
  if (isRequired) {
    assert(
      `Either a block "<:${name}>" or argument "@${name}" are required.`,
      Boolean(blockExists || argument)
    );
  }

  assert(
    `Either a block "<:${name}>" or argument "@${name}" may be provided, but not both`,
    !(blockExists && argument)
  );

 // By returning true here, our if block above should be satisfied
  return true;
};
```


I think by returning a boolean _after_ the assertions run, that should allow us to make this all 

```hbs
{{#if (this.assertBlockOrLabel (hash blockExists=(has-block "label") name="label" argument=@label) isRequired=true)}}
  <span
    class="type-md-tight block leading-4
      {{if @isDisabled 'text-disabled' 'text-titles-and-attributes'}}"
    data-label
  >
    {{#if (has-block "label")}}
      {{yield to="label"}}
    {{else}}
      {{@label}}
    {{/if}}
  </span>
{{/if}}
```